include/stl2/view/drop.hpp

```diff
@@ -47,36 +49,62 @@ STL2_OPEN_NAMESPACE {
 
 			constexpr R base() const { return base_; }
 
-			constexpr auto begin() requires !SimpleView<R> { return begin_impl(*this); }
-			constexpr auto begin() const requires Range<R const>
+			constexpr auto begin()
+			requires !(SimpleView<R> && RandomAccessRange<R>)
+			{ return begin_impl(*this); }
+			constexpr auto begin() const requires Range<R const> && RandomAccessRange<R const>
 			{ return begin_impl(*this); }
 
-			constexpr auto end() requires !SimpleView<R> { return end_impl(*this); }
-			constexpr auto end() const requires Range<R const>
+			constexpr auto end()
+			requires !(SimpleView<R> && RandomAccessRange<R>)
```

Comment:
I suppose the `end` overloads don't need the `RandomAccessRange` constraints, since they don't need to access the cache.

Reply:
Admittedly it could be confusing to have differing requirements on `begin` and `end` - I don't know that I have a preference here.